### DISCUSSION TITLE
Exponential Moving Average/Polyak Averaging in MXNet

### DISCUSSION THREAD

**Discussion by anirudhacharya** [apache/incubator-mxnet#13585] [Q&A]
How can I implement Polyak averaging of as part of my network, in the wavenet model - https://arxiv.org/abs/1711.10433

I am not sure it can be implemented as an optimizer, it is more like a post-processing step on the weights after the optimizer update.

Tensorflow has it here - https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [Feature Request, Question, Optimizer]

**Comment by Ishitori** [comment]
I have an implementation of Polyak averaging. Take a look here to find out if it is the same version they used in WaveNet: https://github.com/dmlc/gluon-nlp/blob/bidaf/scripts/question_answering/utils.py#L277

**Comment by anirudhacharya** [comment]
I will check. Thanks!

**Comment by sojiadeshina** [comment]
@safrooze also implemented Polyak Averaging a la wavenet and may be able to help.
